### PR TITLE
Updated Dockerfile, scripts and documentation to support arcaflow hog scenarios

### PR DESCRIPTION
**Pull request by tsebastiani**

### CODE REVIEW

File: node-cpu-hog/run.sh [hunk 1]
```diff
@@ -2,18 +2,29 @@
 
 set -ex
 
+ROOT_FOLDER="/root"
+KRAKEN_FOLDER="$ROOT_FOLDER/kraken"
+SCENARIO_FOLDER="$KRAKEN_FOLDER/scenarios/arcaflow/memory-hog"
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply node-cpu-hog/run.sh:7]: I am using docker on my mac so not sure if its a environment issue

[the remark this answers is not in the document]
Comment (paigerube14) [reply node-cpu-hog/run.sh:7]: My mistake with some variables

**Comment by paigerube14** [comment]
On both this PR and Ravi's open one I am getting the below error on a 4.14 cluster. 

```
2023-05-22T21:34:09Z info Deploying pod from image quay.io/arcalot/arcaflow-plugin-stressng:latest...

2023-05-22T21:34:09Z error Invalid workflow (invalid workflow (failed to load schema for step workload_loop (invalid workflow (failed to load schema for step stressng (failed to deploy plugin from image quay.io/arcalot/arcaflow-plugin-stressng:latest (failed to create pod (pods "arcaflow-plugin-x22k5" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "arcaflow-plugin-container" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "arcaflow-plugin-container" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "arcaflow-plugin-container" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "arcaflow-plugin-container" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost"))))))))

2023-05-22 21:34:09,797 [ERROR] failed to run arcaflow scenario scenarios/arcaflow/cpu-hog/input.yaml
```

**Comment by chaitanyaenr** [comment]
Code 

> This might be something that we should report for other Arca plugins side as well. cc: @dustinblack

**Comment by chaitanyaenr** [comment]
> On both this PR and Ravi's open one I am getting the below error on a 4.14 cluster.
> 
> ```
> 2023-05-22T21:34:09Z info Deploying pod from image quay.io/arcalot/arcaflow-plugin-stressng:latest...
> 
> 2023-05-22T21:34:09Z error Invalid workflow (invalid workflow (failed to load schema for step workload_loop (invalid workflow (failed to load schema for step stressng (failed to deploy plugin from image quay.io/arcalot/arcaflow-plugin-stressng:latest (failed to create pod (pods "arcaflow-plugin-x22k5" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "arcaflow-plugin-container" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "arcaflow-plugin-container" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "arcaflow-plugin-container" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "arcaflow-plugin-container" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost"))))))))
> 
> 2023-05-22 21:34:09,797 [ERROR] failed to run arcaflow scenario scenarios/arcaflow/cpu-hog/input.yaml
> ```
This might be something that we should report for other Arca plugins side as well. cc: @dustinblack

File: node-cpu-hog/run.sh (hunk 1 above)
Comment (chaitanyaenr) [inline node-cpu-hog/run.sh:7]: s/memory-hog/cpu-hog

**Comment by chaitanyaenr** [comment]
Testing update: CPU-hog works as expected with https://github.com/redhat-chaos/krkn-hub/pull/95#discussion_r1201599152.

**Comment by chaitanyaenr** [comment]
Testing update: Memory hog scenario is working as expected.

File: node-cpu-hog/env.sh [hunk 2]
```diff
@@ -1,13 +1,12 @@
 #!/bin/bash
 
 # Vars and respective defaults
-export JOB_CLEANUP_POLICY=${JOB_CLEANUP_POLICY:="delete"}
-export LITMUS_INSTALL=${LITMUS_INSTALL:=true}
```
Comment (chaitanyaenr) [inline node-cpu-hog/env.sh:5]: We might want to remove Litmus vars from https://github.com/redhat-chaos/krkn-hub/blob/main/env.sh#L14 as well.

**Comment by chaitanyaenr** [comment]
Hitting the following for the IO hog scenario with the defaults + node-selector set:
```
2023-05-23T06:44:10Z	error		Output ID for step "stressng" is "error".
Output data: "map[error:/usr/bin/stress-ng failed with return code
                1:
aborting: temp-path '.' must be readable and writeable
]"
```

**Comment by tsebastiani** [comment]
> On both this PR and Ravi's open one I am getting the below error on a 4.14 cluster.
> 
> ```
> 2023-05-22T21:34:09Z info Deploying pod from image quay.io/arcalot/arcaflow-plugin-stressng:latest...
> 
> 2023-05-22T21:34:09Z error Invalid workflow (invalid workflow (failed to load schema for step workload_loop (invalid workflow (failed to load schema for step stressng (failed to deploy plugin from image quay.io/arcalot/arcaflow-plugin-stressng:latest (failed to create pod (pods "arcaflow-plugin-x22k5" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "arcaflow-plugin-container" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "arcaflow-plugin-container" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "arcaflow-plugin-container" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "arcaflow-plugin-container" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost"))))))))
> 
> 2023-05-22 21:34:09,797 [ERROR] failed to run arcaflow scenario scenarios/arcaflow/cpu-hog/input.yaml
> ```
Opened a PR on the arcaflow kubernetes deployer to support requested securityContext policies once merged I'm gonna add them to the scenarios
https://github.com/arcalot/arcaflow-engine-deployer-kubernetes/pull/13

File: node-cpu-hog/env.sh (hunk 2 above)
Comment (tsebastiani) [reply node-cpu-hog/env.sh:5]: There are several litmus variables around, if you are ok I can remove all of them.

**Comment by tsebastiani** [comment]
> Hitting the following for the IO hog scenario with the defaults + node-selector set:
> 
> ```
> 2023-05-23T06:44:10Z	error		Output ID for step "stressng" is "error".
> Output data: "map[error:/usr/bin/stress-ng failed with return code
>                 1:
> aborting: temp-path '.' must be readable and writeable
> ]"
> ```

This must be related to hostPath permissions on your cluster, In the arcaflow kube deployer mentioned above I've exposed several security context options (including pod level runAsUser, runAsGroup, fsGroup) that may affect and fix this behaviour, but we've to wait that are merged to make an extensive test.

**Comment by chaitanyaenr** [comment]
Testing update: Tried passing multiple node-selectors to launch multiple replicas on the same node, run hog scenarios on different nodes at once and it works as expected.

**Review by paigerube14 — APPROVED** [review]
/lgtm

**Pull request merged**